Assets/Scripts/Gameplay/ConnectionManagement/ConnectionState/ConnectedConnectionState.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using Unity.Multiplayer.Samples.BossRoom.ApplicationLifecycle.Messages;
+using Unity.Multiplayer.Samples.BossRoom.Shared.Infrastructure;
+
+namespace Unity.Multiplayer.Samples.BossRoom
+{
+    /// <summary>
+    /// Connection state corresponding to a connected client. When being disconnected or timed out, transitions to the
+    /// Offline or Reconnecting state, depending on the cause of the disconnect.
+    /// </summary>
+    public class ConnectedConnectionState : ConnectionState
+    {
+        IPublisher<QuitGameSessionMessage> m_QuitGameSessionPublisher;
+        ISubscriber<ConnectStatus> m_ConnectStatusSubscriber;
+        IDisposable m_Subscription;
+
+        ConnectStatus m_ConnectStatus;
+
+        [Inject]
+        void InjectDependencies(IPublisher<QuitGameSessionMessage> quitGameSessionPublisher, ISubscriber<ConnectStatus> connectStatusSubscriber)
+        {
+            m_QuitGameSessionPublisher = quitGameSessionPublisher;
+            m_ConnectStatusSubscriber = connectStatusSubscriber;
+        }
+
+        public override void Enter()
+        {
+            m_Subscription = m_ConnectStatusSubscriber.Subscribe(status => m_ConnectStatus = status);
```

Comment:
why is this handled in this state? Shouldn't this be provided in OnClientDisconnect as a parameter?

Reply:
My intent here was to avoid giving the ConnectionManager this responsibility to hold this data.
I had an alternative solution in mind that could avoid both of those, but I'm not entirely convinced by it. I think it might be easier to discuss it in person, though